core/src/main/java/org/elasticsearch/search/aggregations/AbstractAggregationBuilder.java

```diff
@@ -116,6 +116,11 @@ public AB setMetaData(Map<String, Object> metaData) {
     }
 
     @Override
+    public Map<String, Object> getMetaData() {
```

Comment:
Why does the metaData need to be accessible for anything other than subclasses? My current understanding is that it is used to associate state in the request with the aggregation response, so I'm not sure if we need to expose these properties.